### PR TITLE
Add Shared GC builds to CI

### PR DESCRIPTION
**Pull request by eightbitraptor**
In order to prevent regressions to the GC API that will break existing modular GC's we should add cases to CI that cover building existing known shared GC implementations included with Ruby.

### CODE REVIEW

**Comment by nobu** [comment]
Is `--with-shared-gc` in ubuntu.yml no longer needed?

File: .github/workflows/modgc.yml [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+      - name: Verify MMTk bindgen
+        run: |
+          cd ../src/gc/mmtk
+          cargo install --force cbindgen
+          cbindgen --config cbindgen.toml --output ../mmtk.h
+          if read -n1 -d '' < <(git diff ../mmtk.h); then
+            git diff ../mmtk.h
+            exit 1
+          fi
+        if: ${{ matrix.gc.name == 'mmtk' }}
```
Comment (peterzhu2118) [inline .github/workflows/modgc.yml:150]: We probably don't want to verify the bindgen in this CI.

**Review by peterzhu2118 — COMMENTED** [review]
We probably want to commit the `spec/mmtk.mspec` and `test/.excludes-mmtk` upstream and sync it here.

File: common.mk [hunk 2]
```diff
@@ -753,6 +753,7 @@ clean-extout: PHONY
 	-$(Q)$(RMDIR) $(EXTOUT)/$(arch) $(RUBYCOMMONDIR) $(EXTOUT) 2> $(NULL) || $(NULLCMD)
 clean-gc: PHONY
 	$(Q) $(RMALL) .gc
+	$(Q) $(RMALL) gc
```
[the remark this answers is not in the document]
Comment (peterzhu2118) [reply common.mk:756]: I have a fix at #12295.

[the remark this answers is not in the document]
Comment (rwstauner) [reply common.mk:756]: Thank you!

**Pull request merged**